packages/x-tree-view/src/internals/useTreeView/extractPluginParamsFromProps.ts

```diff
@@ -0,0 +1,76 @@
+import {
+  ConvertSignaturesIntoPlugins,
+  MergeSignaturesProperty,
+  TreeViewAnyPluginSignature,
+  TreeViewPlugin,
+  TreeViewPluginSignature,
+} from '../models';
+import { UseTreeViewBaseProps } from './useTreeView.types';
+import { TreeViewCorePluginSignatures } from '../corePlugins';
+
+interface ExtractPluginParamsFromPropsParameters<
+  TSignatures extends readonly TreeViewAnyPluginSignature[],
+  TProps extends Partial<UseTreeViewBaseProps<TSignatures>>,
+> {
+  plugins: ConvertSignaturesIntoPlugins<readonly [...TreeViewCorePluginSignatures, ...TSignatures]>;
+  props: TProps;
+}
+
+interface ExtractPluginParamsFromPropsReturnValue<
+  TSignatures extends readonly TreeViewAnyPluginSignature[],
+  TProps extends Partial<UseTreeViewBaseProps<TSignatures>>,
+> extends UseTreeViewBaseProps<TSignatures> {
+  pluginParams: MergeSignaturesProperty<TSignatures, 'defaultizedParams'>;
+  forwardedProps: Omit<TProps, keyof MergeSignaturesProperty<TSignatures, 'params'>>;
+}
+
+export const extractPluginParamsFromProps = <
+  TSignatures extends readonly TreeViewPluginSignature<any>[],
+  TProps extends Partial<UseTreeViewBaseProps<TSignatures>>,
+>({
+  props: { slots, slotProps, apiRef, experimentalFeatures, ...props },
+  plugins,
+}: ExtractPluginParamsFromPropsParameters<
+  TSignatures,
+  TProps
+>): ExtractPluginParamsFromPropsReturnValue<TSignatures, TProps> => {
+  type PluginParams = MergeSignaturesProperty<TSignatures, 'params'>;
+
+  const paramsLookup = {} as Record<keyof PluginParams, true>;
+  plugins.forEach((plugin) => {
+    Object.assign(paramsLookup, plugin.params);
+  });
+
+  const pluginParams = {} as PluginParams;
+  const forwardedProps = {} as Omit<TProps, keyof PluginParams>;
+
+  Object.keys(props).forEach((propName) => {
+    const prop = props[propName as keyof typeof props] as any;
```

Comment:
We had similar issues for normalizing plugins in charts without using `any`. If one day you find a better solution than your current one, I would be interested :)

Reply:
![image](https://github.com/mui/mui-x/assets/3309670/b115bc7e-9d33-4152-b182-fac5c2c93ad8)

Given the types of `Object.keys` I think there is no better solution for now :cry: